File: LoginRadiusSDK/LoginRadiusError.m
//
//  NSError+LRError.m
//
//  Copyright © 2016 LoginRadius Inc. All rights reserved.
//

#import "LoginRadiusError.h"

static NSString * _Nonnull const LoginRadiusDomain = @"com.loginradius";

@implementation NSError (LoginRadiusError)

+ (NSError *)errorWithCode:(NSInteger)code
			   description:(NSString *)description
			 failureReason:(NSString *)failureReason {
	NSError *err = [NSError errorWithDomain:LoginRadiusDomain
									   code:code
								   userInfo:@{
											  NSLocalizedDescriptionKey: description,
											  NSLocalizedFailureReasonErrorKey: failureReason,
											  }];
	return err;
}


@end

File: Sources/LoginRadiusError.swift
//
//  LoginRadiusError.swift
//  LoginRadiusSwift SDK
//  Created by Megha Agarwal.
// Copyright © 2023 LoginRadius Inc. All rights reserved.
//

import Foundation

public let LoginRadiusDomain = "com.loginradius"

extension NSError {
    public static func error(withCode code: Int, description: String, failureReason: String) -> NSError {
        let userInfo = [
            NSLocalizedDescriptionKey: description,
            NSLocalizedFailureReasonErrorKey: failureReason
        ]
        return NSError(domain: LoginRadiusDomain, code: code, userInfo: userInfo)
    }
}
